src/GraphQL.Conventions/Extensions/Utilities.cs

```diff
@@ -19,5 +27,82 @@ public static class Utilities
+        /// Ref/out parameters are not supported.
+        /// </param>
+        /// <returns>
+        /// The return value from the method, or null for methods that return void.
+        /// </returns>
+        public static object InvokeEnhanced(this MethodInfo methodInfo, object instance, object[] arguments)
+        {
+            //just good practice
+            if (methodInfo == null)
+                throw new ArgumentNullException(nameof(methodInfo));
+            //the following check is more descriptive than the NullReferenceException that would otherwise occur
+            if (!methodInfo.IsStatic && instance == null)
+                throw new ArgumentNullException(nameof(instance), "Instance is required for static methods");
+            //the lambda ignores extra arguments so the following check is necessary
+            if (methodInfo.GetParameters().Length != (arguments?.Length ?? 0))
+                throw new ArgumentException("Invalid number of arguments for this method", "arguments");
+            //type errors will be thrown as InvalidCastException inside the lambda
+
+            var lambda = _methodDictionary.GetOrAdd(methodInfo, InvokeEnhancedUncached);
+
+            return lambda(instance, arguments);
+        }
+        private static ConcurrentDictionary<MethodInfo, Func<object, object[], object>> _methodDictionary = new ConcurrentDictionary<MethodInfo, Func<object, object[], object>>();
+        private static Func<object, object[], object> InvokeEnhancedUncached(MethodInfo methodInfo)
+        {
+            var instanceParameter = Expression.Parameter(typeof(object));
+            var argumentsParameter = Expression.Parameter(typeof(object[]));
+            var instanceExpression = methodInfo.IsStatic ? null : Expression.Convert(instanceParameter, methodInfo.DeclaringType);
+            var methodParameters = methodInfo.GetParameters();
+            var parameters = methodParameters.Select((param, index) => {
+                return Expression.Convert(Expression.ArrayAccess(argumentsParameter, Expression.Constant(index)), param.ParameterType);
+            });
+            var call = Expression.Call(instanceExpression, methodInfo, parameters);
+            Expression body;
+            if (methodInfo.ReturnType == typeof(void))
+            {
+                body = Expression.Block(new Expression[] {
+                    call,
+                    Expression.Constant(null, typeof(object))
+                });
+            }
+            else
+            {
+                body = Expression.Convert(call, typeof(object));
+            }
+            var lambda = Expression.Lambda<Func<object, object[], object>>(body, instanceParameter, argumentsParameter);
+            return lambda.Compile();
+        }
+
+        public static object InvokeEnhanced(this ConstructorInfo constructorInfo, object[] parameters)
+        {
+            try
+            {
+                return constructorInfo.Invoke(parameters);
```

Comment:
Why do you use invoke here instead of dynamic compilation from previous method?

Reply:
It's just a bunch more work to do it again for constructors, which I have not done.  The method can always be replaced so that exceptions and not rethrown.

Reply:
If it's not too much more work, I'll redo that also.

Reply:
I did it for constructors also.  Mostly copy and paste.  The hardest part was writing a half dozen new tests.